plugins/destination/postgresql/client/migrate.go

```diff
@@ -326,50 +346,51 @@ func (c *Client) addColumn(ctx context.Context, tableName string, column schema.
 	return nil
 }
 
-func (c *Client) createTableIfNotExist(ctx context.Context, table *schema.Table) error {
+func (c *Client) createTableIfNotExist(ctx context.Context, table *arrow.Schema) error {
 	var sb strings.Builder
-	tableName := pgx.Identifier{table.Name}.Sanitize()
+	tName := schema.TableName(table)
+	tableName := pgx.Identifier{tName}.Sanitize()
 	sb.WriteString("CREATE TABLE IF NOT EXISTS ")
 	sb.WriteString(tableName)
 	sb.WriteString(" (")
-	totalColumns := len(table.Columns)
+	totalColumns := len(table.Fields())
 
 	primaryKeys := []string{}
-	for i, col := range table.Columns {
+	for i, col := range table.Fields() {
 		pgType := c.SchemaTypeToPg(col.Type)
 		if pgType == "" {
-			c.logger.Warn().Str("table", table.Name).Str("column", col.Name).Msg("Column type is not supported, skipping")
+			c.logger.Warn().Str("table", tName).Str("column", col.Name).Msg("Column type is not supported, skipping")
 			continue
 		}
 		columnName := pgx.Identifier{col.Name}.Sanitize()
 		fieldDef := columnName + " " + pgType
-		if col.CreationOptions.Unique {
+		if schema.IsUnique(col) {
 			fieldDef += " UNIQUE"
 		}
-		if col.CreationOptions.NotNull {
+		if !col.Nullable {
 			fieldDef += " NOT NULL"
 		}
 		sb.WriteString(fieldDef)
 		if i != totalColumns-1 {
 			sb.WriteString(",")
 		}
-		if c.enabledPks() && col.CreationOptions.PrimaryKey {
+		if c.enabledPks() && schema.IsPk(col) {
 			primaryKeys = append(primaryKeys, col.Name)
 		}
 	}
 
 	if len(primaryKeys) > 0 {
 		// add composite PK constraint on primary key columns
 		sb.WriteString(", CONSTRAINT ")
-		sb.WriteString(table.Name)
+		sb.WriteString(tName)
```

Comment:
```suggestion
		sb.WriteString(tableName)
```

I think we want the sanitized version here